app/core/WalletConnect/WalletConnectV2.test.ts

```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
     });
   });
 
-  describe('WC2Manager connect', () => {
-    it('should handle v2 URIs correctly', async () => {
-      const mockWcUri = 'wc://testv2uri';
+  describe('WC2Manager isWalletConnect', () => {
+    it('should return true for valid WalletConnect origins', () => {
+      const result = manager.isWalletConnect('https://example.com');
+      expect(result).toBe(true);
+    });
 
-      await manager.connect({
-        wcUri: mockWcUri,
-        redirectUrl: 'https://example.com',
-        origin: AppConstants.DEEPLINKS.ORIGIN_DEEPLINK,
+    it('should return false for invalid WalletConnect origins', () => {
+      const result = manager.isWalletConnect('https://invalid.com');
+      expect(result).toBe(false);
+    });
+  });
+
+  describe('WC2Manager session proposal handling', () => {
+    it('should handle session proposal rejection', async () => {
```

Comment:
S: rename to comply to guidelines.
I suggest somehting like this, but please clarify it with the real wording, not sure it's an event, wallet, the right terms...

```suggestion
    it('returns rejectSession event to wallet on proposal rejection', async () => {
```